### PR TITLE
IGNITE-16360 Fix CdcLoader instantiates several Spring contexts

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/client/router/impl/GridRouterCommandLineStartup.java [hunk 1]
```diff
@@ -49,20 +50,21 @@
      *
      * @param beans Beans loaded from spring configuration file.
      */
-    public void start(Map<Class<?>, Object> beans) {
-        log = (IgniteLogger)beans.get(IgniteLogger.class);
-
-        if (log == null) {
+    public void start(Map<Class<?>, Collection> beans) {
+        if (F.isEmpty(beans.get(IgniteLogger.class))) {
```
Comment (ololo3000) [inline modules/core/src/main/java/org/apache/ignite/internal/client/router/impl/GridRouterCommandLineStartup.java:54]: Let's just change 
`log = (IgniteLogger)beans.get(IgniteLogger.class);`
to
`log = F.<IgniteLogger>first(beans.get(IgniteLogger.class));`

And do the same with tcpCfg.

File: modules/core/src/main/java/org/apache/ignite/internal/util/spring/IgniteSpringHelper.java [hunk 2]
```diff
@@ -99,7 +99,10 @@
      *       map value is {@code null}.
      * @throws IgniteCheckedException If failed to load configuration.
      */
-    public Map<Class<?>, Object> loadBeans(URL cfgUrl, Class<?>... beanClasses) throws IgniteCheckedException;
+    public IgniteBiTuple<Map<Class<?>, Collection>, ? extends GridSpringResourceContext> loadBeans(
```
Comment (ololo3000) [inline modules/core/src/main/java/org/apache/ignite/internal/util/spring/IgniteSpringHelper.java:102]: It seems that we should update the JavaDoc.

File: modules/spring/src/main/java/org/apache/ignite/internal/util/spring/IgniteSpringHelperImpl.java [hunk 3]
```diff
@@ -145,17 +143,24 @@
...
 
         ApplicationContext springCtx = initContext(cfgUrl);
 
-        Map<Class<?>, Object> beans = new HashMap<>();
+        Map<Class<?>, Collection> beans = new HashMap<>();
 
-        for (Class<?> cls : beanClasses)
-            beans.put(cls, bean(springCtx, cls));
+        for (Class<?> cls : beanClasses) {
+            Map<String, ?> b = springCtx.getBeansOfType(cls);
```
Comment (ololo3000) [inline modules/spring/src/main/java/org/apache/ignite/internal/util/spring/IgniteSpringHelperImpl.java:157]: Let's rename `b` to `clsBeans`.

File: modules/core/src/main/java/org/apache/ignite/internal/client/router/impl/GridRouterCommandLineStartup.java (hunk 1 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/client/router/impl/GridRouterCommandLineStartup.java:54]: Fixed

**Review by ololo3000 — APPROVED** [review]

**Pull request merged**